sdk/src/azure/core/az_context.c

```diff
@@ -33,6 +35,8 @@ AZ_NODISCARD int64_t az_context_get_expiration(az_context const* context)
 AZ_NODISCARD az_result
 az_context_get_value(az_context const* context, void const* key, void const** out_value)
 {
+  _az_PRECONDITION_NOT_NULL(context);
```

Comment:
Key can't be null either. Add that precondition.

Also, add a runtime check, if key is null, return `AZ_ERROR_ITEM_NOT_FOUND`, for when preconditions are off.

Also, tests :)

Reply:
If its a precondition it shouldn't be a runtime check.  The precondition is the equivalent of an `if(NULL)` check.